### DISCUSSION TITLE
Change the aggregation function from sum to average

### DISCUSSION THREAD

**Discussion by arashkav** [visgl/flowmap.gl#20] [Q&A] [answered]
Is there any way that we change aggregation function from count (summing up values) when clustering happens to avg or other functions?

**Comment by ilyabo** [comment, marked as answer]
Not at the moment, but it should be possible to make the aggregation function pluggable. If you want to submit a PR, I would be happy to review. The relevant code is [here](https://github.com/visgl/flowmap.gl/blob/d7465fd93c5a7717ca5d58bc84dafc4a88587554/packages/data/src/cluster/ClusterIndex.ts#L181). Not quite sure though how it should work for the location totals.

**Reply by arashkav** [reply]
Thanks! That would be great and open up a lot of other applications. Do you want me to submit it as an <<idea>>?

**Reply by ilyabo** [reply]
Go ahead.

**Comment by jtbaker** [comment]
Related PR for anyone that comes across this: https://github.com/visgl/flowmap.gl/pull/22
